zipline/data/fx/hdf5.py

```diff
@@ -203,56 +201,55 @@ def get_rates(self, rate, quote, bases, dts):
         #
         # However, we also need to handle two important edge cases:
         #
-        #   1. row_ixs contains -1 for dts before the start of self.dts.
-        #   2. col_ixs contains -1 for any currencies we don't know about.
+        #   1. row_ixs contains -1 for any currencies we don't know about.
+        #   2. col_ixs contains -1 for dts before the start of self.dts.
         #
         # If either of the above cases obtains, we want to return NaN for the
         # corresponding output locations.
 
         # We handle (1) by reading raw data into a buffer with one extra
-        # row. When we then apply the row index to permute the raw data into
-        # the correct order, any rows with values of -1 will pull from the
-        # extra row, which will always contain NaN>
+        # column. When we then apply the column index to permute the raw data
+        # into the correct order, any columnss with values of -1 will pull from
+        # the extra column, which will always contain NaN>
         #
-        # We handle (2) by overwriting columns with indices of -1 with NaN as a
+        # We handle (2) by overwriting rows with indices of -1 with NaN as a
         # postprocessing step.
-        slice_begin = max(row_ixs[0], 0)
-        slice_end = max(row_ixs[-1], 0) + 1  # +1 to be inclusive of end date.
+        slice_begin = max(col_ixs[0], 0)
+        slice_end = max(col_ixs[-1], 0) + 1  # +1 to be inclusive of end date.
 
-        # Allocate a buffer full of NaNs with one extra row/column. See
+        # Allocate a buffer full of NaNs with one extra column. See
         # OPTIMIZATION notes above.
         buf = np.full(
-            (slice_end - slice_begin + 1, len(self.currencies)),
+            (len(self.currencies), slice_end - slice_begin + 1),
             np.nan,
         )
 
-        # Read data into all but the last row/column of the buffer.
-        dataset.read_direct(
-            buf[:-1],
-            np.s_[slice_begin:slice_end],
-        )
+        buf[:, :-1] = dataset[:, slice_begin:slice_end]
 
         # Permute the rows into place, pulling from the empty NaN locations for
-        # row/column indices of -1.
-        out = buf[:, col_ixs][row_ixs - slice_begin]
+        # column indices of -1.
+        out = buf[:, col_ixs - slice_begin][row_ixs]
 
-        # Fill missing columns with NaN. See OPTIMIZATION notes above.
-        out[:, col_ixs == -1] = np.nan
+        # Fill missing rows with NaN. See OPTIMIZATION notes above.
+        out[row_ixs == -1] = np.nan
 
-        return out
+        # Transpose everything to maintain dts as row labels, currencies as col
+        # labels which is expected everywhere else.
+        return out.transpose()
 
 
 class HDF5FXRateWriter(object):
     """Writer class for HDF5 files consumed by HDF5FXRateReader.
     """
-    def __init__(self, group):
+    def __init__(self, group, date_chunk_size):
```

Comment:
I think it would probably be reasonable to give this a default value so that zipline users don't need to do the same tuning work that we did on this.

Reply:
I added the default as a global defined at the beginning of the file.